dbms/src/TableFunctions/TableFunctionRandom.cpp

```diff
@@ -0,0 +1,477 @@
+#include <Common/Exception.h>
+
+#include <Core/Block.h>
+#include <Storages/StorageValues.h>
+#include <DataTypes/DataTypeTuple.h>
+#include <DataTypes/DataTypeEnum.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypeDateTime64.h>
+#include <DataTypes/DataTypeDecimalBase.h>
+#include <DataTypes/DataTypeArray.h>
+#include <Columns/ColumnArray.h>
+#include <Columns/ColumnFixedString.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnVector.h>
+#include <Columns/ColumnNullable.h>
+#include <Columns/ColumnTuple.h>
+
+
+#include <Parsers/ASTExpressionList.h>
+#include <Parsers/ASTLiteral.h>
+#include <Parsers/ASTFunction.h>
+
+#include <Common/randomSeed.h>
+#include <pcg_random.hpp>
+
+#include <TableFunctions/ITableFunction.h>
+#include <TableFunctions/TableFunctionRandom.h>
+#include <TableFunctions/TableFunctionFactory.h>
+#include <TableFunctions/parseColumnsListForTableFunction.h>
+
+#include "registerTableFunctions.h"
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int BAD_ARGUMENTS;
+    extern const int BAD_TYPE_OF_FIELD;
+    extern const int LOGICAL_ERROR;
+}
+
+void fillColumnWithRandomData(IColumn & column, DataTypePtr type, UInt64 limit,
+                              UInt64 max_array_length, UInt64 max_string_length, UInt64 random_seed)
+{
+    TypeIndex idx = type->getTypeId();
+    if (!random_seed)
+        random_seed = randomSeed();
+    (void) max_string_length;
+
+    switch (idx)
+    {
+        case TypeIndex::Nothing:
+            throw Exception("Random Generator not implemented for type 'Nothing'.", ErrorCodes::NOT_IMPLEMENTED);
+        case TypeIndex::UInt8:
+        {
+            auto & data = typeid_cast<ColumnVector<UInt8> &>(column).getData();
+            data.resize(limit);
+            pcg32 generator(random_seed);
+            for (UInt64 i = 0; i < limit; ++i)
+            {
+                data[i] = static_cast<UInt8>(generator());
+            }
+            break;
+        }
+        case TypeIndex::UInt16:
+        {
+            auto & data = typeid_cast<ColumnVector<UInt16> &>(column).getData();
+            data.resize(limit);
+            pcg32 generator(random_seed);
+            for (UInt64 i = 0; i < limit; ++i)
+            {
+                data[i] = static_cast<UInt16>(generator());
+            }
+            break;
+        }
+        case TypeIndex::UInt32:
+        {
+            auto & data = typeid_cast<ColumnVector<UInt32> &>(column).getData();
+            data.resize(limit);
+            pcg32 generator(random_seed);
+            for (UInt64 i = 0; i < limit; ++i)
+            {
+                data[i] = static_cast<UInt32>(generator());
+            }
+            break;
+        }
+        case TypeIndex::UInt64:
+        {
+            auto & data = typeid_cast<ColumnVector<UInt64> &>(column).getData();
+            data.resize(limit);
+            pcg32 generator(random_seed);
```

Comment:
Why don't just use pcg64?

Reply:
It is not stable even with predefined seed.

Reply:
Don't believe it's possible.

Reply:
Probably some other bug interfered with this. I had different results on Mac and Linux. Currently everything is OK